reactioncommerce/reaction
# Fix 3617 Marketplace - allow users to become sellers

Closes #3617 

### To test 
1. Run `reaction`
1. Login as admin, enable Marketplace, check checkbox to allow users to register as sellers
1. Logout and register a new account
1. Visit profile, click "Become a seller"
1. Observe no error and everything working.

### To review

The bug was that the when a new shop was getting added it wasn't published to the client. This is because Meteor doesn't re-execute the server's publisher code again. 
So now a reactive list of user's shops is passed to the publisher so that whenever user's shops change, the list is published again.

I tried using observing whole of `Shops` collection with `observe()/added()` but when a new shop is added, first the shop is added and then permissions are added to user. So I wan't able to check what permissions the user had for the new shop added.
